src/commands/cmd_stream.cc

```diff
@@ -1635,6 +1635,14 @@ class CommandXReadGroup : public Commander,
     redis::Stream stream_db(srv_->storage, conn_->GetNamespace());
 
     std::vector<StreamReadResult> results;
+
+    std::vector<std::string> lock_keys;
+    lock_keys.reserve(streams_.size());
+    for (auto &stream_name : streams_) {
+      auto ns_key = stream_db.AppendNamespacePrefix(stream_name);
+      lock_keys.emplace_back(std::move(ns_key));
+    }
+    MultiLockGuard guard(srv_->storage->GetLockManager(), lock_keys);
```

Comment:
I think there's two `OnWrite` in stream? one for `XREADGROUP` and another for `XREAD`. cc @torwig

Reply:
I'm not very familiar with the stream, but I observed that `XREAD` is intended to be read-only. It appears there is no locking mechanism involved, so I refrained from modifying its `OnWrite`.
```
MakeCmdAttr<CommandXRead>("xread", -4, "read-only", NO_KEY),
MakeCmdAttr<CommandXReadGroup>("xreadgroup", -7, "write", NO_KEY),
```

Reply:
Ahh got it.